This adds a necessary compile define to OpenSim's sources so that it
can be compiled into libosim (static library).

---

diff --git a/third_party/libosim/opensim-core/OpenSim/Actuators/osimActuatorsDLL.h b/third_party/libosim/opensim-core/OpenSim/Actuators/osimActuatorsDLL.h
index fae5000e0..2a2bd6e59 100644
--- a/third_party/libosim/opensim-core/OpenSim/Actuators/osimActuatorsDLL.h
+++ b/third_party/libosim/opensim-core/OpenSim/Actuators/osimActuatorsDLL.h
@@ -38,10 +38,12 @@
 #define WIN32_LEAN_AND_MEAN
 #define NOMINMAX
 #include <windows.h>
-#ifdef OSIMACTUATORS_EXPORTS
-#define OSIMACTUATORS_API __declspec(dllexport)
+#if defined(OPENSIM_USE_STATIC_LIBRARIES)
+    #define OSIMACTUATORS_API
+#elif defined (OSIMACTUATORS_EXPORTS)
+    #define OSIMACTUATORS_API __declspec(dllexport)
 #else
-#define OSIMACTUATORS_API __declspec(dllimport)
+    #define OSIMACTUATORS_API __declspec(dllimport)
 #endif
 
 #endif // PLATFORM
diff --git a/third_party/libosim/opensim-core/OpenSim/Analyses/osimAnalysesDLL.h b/third_party/libosim/opensim-core/OpenSim/Analyses/osimAnalysesDLL.h
index 67fabc33d..94f4ed209 100644
--- a/third_party/libosim/opensim-core/OpenSim/Analyses/osimAnalysesDLL.h
+++ b/third_party/libosim/opensim-core/OpenSim/Analyses/osimAnalysesDLL.h
@@ -34,10 +34,12 @@
 #define WIN32_LEAN_AND_MEAN
 #define NOMINMAX
 #include <windows.h>
-#ifdef OSIMANALYSES_EXPORTS
-#define OSIMANALYSES_API __declspec(dllexport)
+#if defined(OPENSIM_USE_STATIC_LIBRARIES)
+    #define OSIMANALYSES_API
+#elif defined(OSIMANALYSES_EXPORTS)
+    #define OSIMANALYSES_API __declspec(dllexport)
 #else
-#define OSIMANALYSES_API __declspec(dllimport)
+    #define OSIMANALYSES_API __declspec(dllimport)
 #endif
 
 #endif // PLATFORM
diff --git a/third_party/libosim/opensim-core/OpenSim/Common/osimCommonDLL.h b/third_party/libosim/opensim-core/OpenSim/Common/osimCommonDLL.h
index fec242da3..460649499 100644
--- a/third_party/libosim/opensim-core/OpenSim/Common/osimCommonDLL.h
+++ b/third_party/libosim/opensim-core/OpenSim/Common/osimCommonDLL.h
@@ -34,7 +34,9 @@
     #define WIN32_LEAN_AND_MEAN
     #define NOMINMAX
     #include <windows.h>
-    #ifdef OSIMCOMMON_EXPORTS
+    #if defined(OPENSIM_USE_STATIC_LIBRARIES)
+        #define OSIMCOMMON_API
+    #elif defined(OSIMCOMMON_EXPORTS)
         #define OSIMCOMMON_API __declspec(dllexport)
     #else
         #define OSIMCOMMON_API __declspec(dllimport)
diff --git a/third_party/libosim/opensim-core/OpenSim/ExampleComponents/osimExampleComponentsDLL.h b/third_party/libosim/opensim-core/OpenSim/ExampleComponents/osimExampleComponentsDLL.h
index c14297287..4465bcef9 100644
--- a/third_party/libosim/opensim-core/OpenSim/ExampleComponents/osimExampleComponentsDLL.h
+++ b/third_party/libosim/opensim-core/OpenSim/ExampleComponents/osimExampleComponentsDLL.h
@@ -33,10 +33,13 @@
 #define WIN32_LEAN_AND_MEAN
 #define NOMINMAX
 #include <windows.h>
-#ifdef OSIMEXAMPLECOMPONENTS_EXPORTS
-#define OSIMEXAMPLECOMPONENTS_API __declspec(dllexport)
+
+#if defined(OPENSIM_USE_STATIC_LIBRARIES)
+    #define OSIMEXAMPLECOMPONENTS_API
+#elif defined(OSIMEXAMPLECOMPONENTS_EXPORTS)
+    #define OSIMEXAMPLECOMPONENTS_API __declspec(dllexport)
 #else
-#define OSIMEXAMPLECOMPONENTS_API __declspec(dllimport)
+    #define OSIMEXAMPLECOMPONENTS_API __declspec(dllimport)
 #endif
 
 #endif // PLATFORM
diff --git a/third_party/libosim/opensim-core/OpenSim/Simulation/osimSimulationDLL.h b/third_party/libosim/opensim-core/OpenSim/Simulation/osimSimulationDLL.h
index 467d9b5d8..7c1dd2ab9 100644
--- a/third_party/libosim/opensim-core/OpenSim/Simulation/osimSimulationDLL.h
+++ b/third_party/libosim/opensim-core/OpenSim/Simulation/osimSimulationDLL.h
@@ -38,10 +38,12 @@
 #define WIN32_LEAN_AND_MEAN
 #define NOMINMAX
 #include <windows.h>
-#ifdef OSIMSIMULATION_EXPORTS
-#define OSIMSIMULATION_API __declspec(dllexport)
+#if defined(OPENSIM_USE_STATIC_LIBRARIES)
+    #define OSIMSIMULATION_API
+#elif defined(OSIMSIMULATION_EXPORTS)
+    #define OSIMSIMULATION_API __declspec(dllexport)
 #else
-#define OSIMSIMULATION_API __declspec(dllimport)
+    #define OSIMSIMULATION_API __declspec(dllimport)
 #endif
 
 #endif // PLATFORM
diff --git a/third_party/libosim/opensim-core/OpenSim/Tools/osimToolsDLL.h b/third_party/libosim/opensim-core/OpenSim/Tools/osimToolsDLL.h
index 402c688b2..5deef7331 100644
--- a/third_party/libosim/opensim-core/OpenSim/Tools/osimToolsDLL.h
+++ b/third_party/libosim/opensim-core/OpenSim/Tools/osimToolsDLL.h
@@ -37,10 +37,12 @@
 #define WIN32_LEAN_AND_MEAN
 #define NOMINMAX
 #include <windows.h>
-#ifdef OSIMTOOLS_EXPORTS
-#define OSIMTOOLS_API __declspec(dllexport)
+#if defined(OPENSIM_USE_STATIC_LIBRARIES)
+    #define OSIMTOOLS_API
+#elif defined(OSIMTOOLS_EXPORTS)
+    #define OSIMTOOLS_API __declspec(dllexport)
 #else
-#define OSIMTOOLS_API __declspec(dllimport)
+    #define OSIMTOOLS_API __declspec(dllimport)
 #endif
 
 #endif // PLATFORM
